### ISSUE TITLE
geth "starting work on payload" resulting to a missed block

### ISSUE THREAD

**Issue by rayanebel** [ethereum/go-ethereum#28174]
## System information

Our Ethereum node is using the following components:
* Teku `23.8.0-jdk17` for the consensus layer
* Geth `v1.12.2` for the execution layer

This components are running inside a Kubernetes cluster.

## Actual behaviour

On Wednesday 7 April, between 14:24 and 14:25 p.m _(UTC)_, an issue occured in our execution layer resulting in a missed block _(Sep-20-2023 14:24:11 UTC - slot 7366319)_.

After digging in the logs of the execution layer, we saw that everything was working and then we start seeing non-habitual messages at the time when the block was missed:

```bash
INFO [09-20|14:22:12.033] Imported new potential chain segment     number=18,177,667 hash=6eca65..87aa20 blocks=1 txs=142  mgas=14.316 elapsed=93.983ms     mgasps=152.328  dirty=1022.70MiB
INFO [09-20|14:22:12.245] Chain head was updated                   number=18,177,667 hash=6eca65..87aa20 root=a873d0..7d0b43 elapsed=3.493088ms
INFO [09-20|14:22:25.417] Imported new potential chain segment     number=18,177,668 hash=2f12bb..d8f698 blocks=1 txs=172  mgas=18.801 elapsed=100.657ms    mgasps=186.781  dirty=1022.78MiB
INFO [09-20|14:22:25.462] Chain head was updated                   number=18,177,668 hash=2f12bb..d8f698 root=0b79c7..e8bcdc elapsed=4.6644ms
INFO [09-20|14:22:36.626] Imported new potential chain segment     number=18,177,669 hash=bc1230..8e8b72 blocks=1 txs=114  mgas=13.308 elapsed=79.835ms     mgasps=166.687  dirty=1022.32MiB
INFO [09-20|14:22:36.662] Chain head was updated                   number=18,177,669 hash=bc1230..8e8b72 root=cc4ea4..464d63 elapsed=2.06282ms
INFO [09-20|14:22:49.444] Imported new potential chain segment     number=18,177,670 hash=78d349..061e6d blocks=1 txs=174  mgas=16.132 elapsed=97.279ms     mgasps=165.827  dirty=1023.10MiB
INFO [09-20|14:22:49.456] Chain head was updated                   number=18,177,670 hash=78d349..061e6d root=adc2da..5142f2 elapsed=3.006108ms
INFO [09-20|14:23:01.427] Imported new potential chain segment     number=18,177,671 hash=a770fc..470ebd blocks=1 txs=141  mgas=12.061 elapsed=97.812ms     mgasps=123.307  dirty=1022.48MiB
INFO [09-20|14:23:01.454] Chain head was updated                   number=18,177,671 hash=a770fc..470ebd root=365f42..818ff0 elapsed=2.700001ms
INFO [09-20|14:23:12.994] Imported new potential chain segment     number=18,177,672 hash=89f4d8..2584ca blocks=1 txs=150  mgas=12.809 elapsed=118.436ms    mgasps=108.150  dirty=1022.45MiB
INFO [09-20|14:23:13.129] Chain head was updated                   number=18,177,672 hash=89f4d8..2584ca root=d39d41..240e3d elapsed=2.311404ms
INFO [09-20|14:23:24.370] Imported new potential chain segment     number=18,177,673 hash=5f40e5..88ded5 blocks=1 txs=221  mgas=15.481 elapsed=255.761ms    mgasps=60.531   dirty=1023.33MiB
INFO [09-20|14:23:24.387] Chain head was updated                   number=18,177,673 hash=5f40e5..88ded5 root=8f81f0..8e2a15 elapsed=7.770819ms
WARN [09-20|14:23:33.656] Peer delivering stale transactions       peer=ee9dec8dabcf344254fdb7ecbd3990fde591ea5e0763b857ae3be43659f19c42 rejected=41
INFO [09-20|14:23:47.785] Imported new potential chain segment     number=18,177,674 hash=969216..439fe8 blocks=1 txs=212  mgas=15.231 elapsed=129.457ms    mgasps=117.653  dirty=1022.10MiB
INFO [09-20|14:23:47.808] Chain head was updated                   number=18,177,674 hash=969216..439fe8 root=ec59f1..f6970f elapsed=4.323262ms

INFO [09-20|14:24:03.085] Starting work on payload                 id=0xb5a863509a649a2d
INFO [09-20|14:24:03.149] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=c7676a..428d56 txs=150  withdrawals=16 gas=13,572,373 fees=0.02325886562   root=f50817..afa9e2 elapsed=63.098ms
INFO [09-20|14:24:05.216] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=b82b0b..932f71 txs=161  withdrawals=16 gas=14,818,054 fees=0.02470676341   root=ceb3e8..b7f735 elapsed=66.760ms
INFO [09-20|14:24:07.290] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=b50ad6..9a18af txs=186  withdrawals=16 gas=16,276,726 fees=0.02538676769   root=45da0d..94e3c3 elapsed=73.927ms
INFO [09-20|14:24:09.537] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=421326..521a66 txs=201  withdrawals=16 gas=18,038,543 fees=0.02732417404   root=d54204..59fb73 elapsed=243.670ms
INFO [09-20|14:24:11.740] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=def1ef..b89daf txs=226  withdrawals=16 gas=20,077,913 fees=0.02997346411   root=55b0b4..377271 elapsed=203.620ms
INFO [09-20|14:24:13.837] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=f8dcdc..68b81b txs=246  withdrawals=16 gas=21,819,489 fees=0.03057814317   root=7477f7..b48152 elapsed=95.507ms
INFO [09-20|14:24:15.086] Stopping work on payload                 id=0xb5a863509a649a2d reason=timeout

INFO [09-20|14:24:17.175] Persisted trie from memory database      nodes=24,518,743 size=5.98GiB time=2m11.737215178s gcnodes=95,341,948 gcsize=36.49GiB gctime=6m55.115725097s livenodes=427,921 livesize=154.28MiB
INFO [09-20|14:24:17.176] Imported new potential chain segment     number=18,177,675 hash=3a143b..9ed685 blocks=1 txs=142  mgas=15.385 elapsed=17.162s      mgasps=0.896    dirty=197.10MiB
WARN [09-20|14:24:17.178] Ignoring already known beacon payload    number=18,177,675 hash=3a143b..9ed685 age=18s
INFO [09-20|14:24:17.199] Chain head was updated                   number=18,177,675 hash=3a143b..9ed685 root=9cebf0..632887 elapsed=3.367004ms
INFO [09-20|14:24:24.177] Imported new potential chain segment     number=18,177,676 hash=215fbe..6e1672 blocks=1 txs=225  mgas=19.230 elapsed=110.319ms    mgasps=174.311  dirty=199.25MiB
INFO [09-20|14:24:24.317] Chain head was updated                   number=18,177,676 hash=215fbe..6e1672 root=64348a..f7ad3e elapsed=2.826243ms
INFO [09-20|14:24:35.925] Imported new potential chain segment     number=18,177,677 hash=49c715..de50fd blocks=1 txs=117  mgas=14.837 elapsed=66.109ms     mgasps=224.436  dirty=200.08MiB
INFO [09-20|14:24:36.177] Chain head was updated                   number=18,177,677 hash=49c715..de50fd root=3d6d15..0a6e25 elapsed=1.869486ms
INFO [09-20|14:24:48.770] Imported new potential chain segment     number=18,177,678 hash=ded3ee..db0fbd blocks=1 txs=119  mgas=17.273 elapsed=65.198ms     mgasps=264.930  dirty=201.08MiB
INFO [09-20|14:24:48.849] Chain head was updated                   number=18,177,678 hash=ded3ee..db0fbd root=006453..d9b77e elapsed=2.393906ms
INFO [09-20|14:25:00.489] Imported new potential chain segment     number=18,177,679 hash=5666c2..64babd blocks=1 txs=117  mgas=7.675  elapsed=58.363ms     mgasps=131.510  dirty=201.74MiB
INFO [09-20|14:25:00.602] Chain head was updated                   number=18,177,679 hash=5666c2..64babd root=8cfe0b..32b377 elapsed=1.395276ms
INFO [09-20|14:25:12.860] Imported new potential chain segment     number=18,177,680 hash=a0a4e6..06b43e blocks=1 txs=202  mgas=21.041 elapsed=134.429ms    mgasps=156.518  dirty=203.45MiB
INFO [09-20|14:25:12.954] Chain head was updated                   number=18,177,680 hash=a0a4e6..06b43e root=02c1e7..d26b34 elapsed=3.825343ms
INFO [09-20|14:25:24.877] Imported new potential chain segment     number=18,177,681 hash=c1d8d7..265410 blocks=1 txs=118  mgas=9.355  elapsed=79.173ms     mgasps=118.163  dirty=204.11MiB
INFO [09-20|14:25:24.928] Chain head was updated                   number=18,177,681 hash=c1d8d7..265410 root=f1825e..73ee31 elapsed=1.337415ms
INFO [09-20|14:25:36.928] Imported new potential chain segment     number=18,177,682 hash=b3c915..55719e blocks=1 txs=129  mgas=9.860  elapsed=102.624ms    mgasps=96.082   dirty=204.83MiB
INFO [09-20|14:25:37.055] Chain head was updated                   number=18,177,682 hash=b3c915..55719e root=dcf37c..ed81ed elapsed=1.632642ms
```

When we take a look at beacon node logs we see an error that indicates a problem at the execution client level.

```bash
2023-09-20 14:21:27.001 INFO  - Slot Event  *** Slot: 7366305, Block: 7e2545980b70ab0c007b39bb1c625efa9de951a81a1f0ddaf6c90ff05b415d4f, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:21:39.001 INFO  - Slot Event  *** Slot: 7366306, Block: 8f3e7fd5aaef88e60b7f53f4ad55b174cc771f58dd578f421791d361eda57d71, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:21:51.000 INFO  - Slot Event  *** Slot: 7366307, Block: 954ec802acb27ed9a67b40da939d05b11bc59cf1e1dbd0ddb66425a63f840a43, Justified: 230196, Finalized: 230195, Peers: 99
2023-09-20 14:22:03.000 INFO  - Slot Event  *** Slot: 7366308, Block: 64b64311aa8127c54de89829ac7979ae1228e46e05c973000e6ff5f166878af5, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:22:15.000 INFO  - Slot Event  *** Slot: 7366309, Block: 7a89dea8f39a390f10ed7d197ff974cb9ab53f2e29f4866ef37935e8201a646d, Justified: 230196, Finalized: 230195, Peers: 99
2023-09-20 14:22:27.000 INFO  - Slot Event  *** Slot: 7366310, Block: 78a3d64e236d403016b1ad0c54910da3e15f83784166f8076a300cf160231127, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:22:39.001 INFO  - Slot Event  *** Slot: 7366311, Block: dd2485359a2167a9aecb8486c447b27f51ff758d4e82842ca54eb79cd14fb5fd, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:22:51.001 INFO  - Slot Event  *** Slot: 7366312, Block: b208370f83a360335c80c126c3caa658ed865456b269b605ed246c605be8d9f0, Justified: 230196, Finalized: 230195, Peers: 99
2023-09-20 14:23:03.001 INFO  - Slot Event  *** Slot: 7366313, Block: 3288baf3d1b2a3f9a8bc814699bca799082a1a990938030734b90af921211a75, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:23:15.000 INFO  - Slot Event  *** Slot: 7366314, Block: 66f6261408e29891212bd0f25f1d44595745cd6d9e668fdb41987a26af5073ce, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:23:27.000 INFO  - Slot Event  *** Slot: 7366315, Block: 3e0e69adfcacd54ac2809cac454e3680ffdfc5bd73da0b011fa5fc342b2d168d, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:23:39.084 INFO  - Slot Event  *** Slot: 7366316, Block:                                                        ... empty, Justified: 230196, Finalized: 230195, Peers: 99
2023-09-20 14:23:51.000 INFO  - Slot Event  *** Slot: 7366317, Block: ce87a52a8f87fa997a29b68cafcbf85ffa2ea4aaff793e4eb7ce70dd5e343dd5, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:24:03.083 INFO  - Slot Event  *** Slot: 7366318, Block:                                                        ... empty, Justified: 230196, Finalized: 230195, Peers: 100

2023-09-20 14:24:03.084 INFO  - Calling local execution layer to start block production (block slot: 7366319)
2023-09-20 16:24:08.003	
2023-09-20 14:24:08.002 WARN  - Execution Client request timed out. Make sure the Execution Client is online and can respond to requests.
2023-09-20 14:24:08.004 ERROR - Unable to import block: Execution Client returned an error: tech.pegasys.teku.ethereum.executionclient.response.InvalidRemoteResponseException: Invalid remote response from the execution client: TimeoutException
2023-09-20 14:24:08.004 WARN  - Late Block Import *** Block: 992e19a9d1c8cc58fb88e39493ca2ec7370700b86b976dc43e7f82dcd74e71bb (7366318) proposer 377897 arrival 995ms, pre-state_retrieved +5ms, processed +165ms, execution_payload_result_received +7838ms, begin_importing +0ms, completed +1ms
2023-09-20 14:24:10.004 INFO  - Retrying execution of block 992e19a9d1c8cc58fb88e39493ca2ec7370700b86b976dc43e7f82dcd74e71bb (7366318)
2023-09-20 14:24:11.004 INFO  - Creating unsigned block for slot 7366319
2023-09-20 14:24:11.042 INFO  - Syncing     *** Target slot: 7366319, Head slot: 7366317, Remaining slots: 2, Connected peers: 101

2023-09-20 14:24:11.043 INFO  - Execution Client is responding to requests again after a previous failure
2023-09-20 14:24:27.001 INFO  - Slot Event  *** Slot: 7366320, Block: 146dee005030e99f0c0a635d711e7bd49c4f7c54839a6800ea546f5adc1765e1, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:24:39.000 INFO  - Slot Event  *** Slot: 7366321, Block: a85b02b52ac2825f3a1b736f6d88af86ad5d44b840f17a13977803523980e2e6, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:24:51.000 INFO  - Slot Event  *** Slot: 7366322, Block: 1605480e21d8c9e8095d99f355cfd9533662f8648609ee1cc5af301b4140c1b2, Justified: 230196, Finalized: 230195, Peers: 99
2023-09-20 14:25:03.001 INFO  - Slot Event  *** Slot: 7366323, Block: d4d3c201a1a0ba544a8c524c5bcd2f93711aa0b59ca880aa345409422694444d, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:25:15.001 INFO  - Slot Event  *** Slot: 7366324, Block: 0c69bcbce670aea991304b0b50424b319b11282d4c977230f9dc96bdcbeb6768, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:25:27.000 INFO  - Slot Event  *** Slot: 7366325, Block: df1d5d9e67ce3988031e5a55012a18f716cf4efc94bd245f684574bdef41426c, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:25:39.001 INFO  - Slot Event  *** Slot: 7366326, Block: 29c34509e8c31cd9905d4303c24fbba0707a5bfe5de74e1c771aae19fdc8d16b, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:25:51.000 INFO  - Slot Event  *** Slot: 7366327, Block: 6621d47dba2e4c6ddcdbe14f4a72c2d8f6e67d320f6979331e1b6d6f37caa88f, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:26:03.000 INFO  - Slot Event  *** Slot: 7366328, Block: 292bbd9585472bdc9230ccc76cb796277b734ea5ba694dbaa3b565d2372eb469, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:26:15.000 INFO  - Slot Event  *** Slot: 7366329, Block: a132b0a1b1f725cf7067c72a420e85dec3450f02101832d6d18b0ae37d98054a, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:26:27.000 INFO  - Slot Event  *** Slot: 7366330, Block: 412a14e73d2fd1219c7aff45d7a91fa5318baa6e2f5c997840a1eef548a4c558, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:26:39.000 INFO  - Slot Event  *** Slot: 7366331, Block: 433fe8d9438f5f47b7c984ec3260c56c6067941ef576e463a1c8cf4b9839fc56, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:26:51.000 INFO  - Slot Event  *** Slot: 7366332, Block: 64cbe0f2a78e3bb2a943e23bf782bca2572b784367ae88b499f8aba3d63b168d, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:27:03.001 INFO  - Slot Event  *** Slot: 7366333, Block: bbbabfb7a018771dbf8abb360ba93f2d4be0bb7f7aab21e9681e68a6855bb411, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:27:15.001 INFO  - Slot Event  *** Slot: 7366334, Block: 9d36690102d0c4726c0abb745afd052074ec821a32197598d892cec7cdf1ded0, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:27:27.001 INFO  - Slot Event  *** Slot: 7366335, Block: 0553400219b3c960b31f67cbde0f856a05354ea3813b4b2769b055c23a856306, Justified: 230196, Finalized: 230195, Peers: 100
2023-09-20 14:27:35.000 INFO  - Epoch Event *** Epoch: 230198, Justified checkpoint: 230197, Finalized checkpoint: 230196, Finalized root: 237a72e48c4142c3ca0e2ebdfbfde34edddb7e603ee61ed06e41b17545e9d56e
2023-09-20 14:27:35.001 INFO  - Block performance: epoch 230197, expected 1, produced 0, included 0 (0%)
2023-09-20 14:27:35.001 INFO  - Sync committee performance: epoch 230197, expected 96, produced 96, correct 90, included 87 (91%)
2023-09-20 14:27:36.498 INFO  - Attestation performance: epoch 230196, expected 3927, produced 3927, included 3927 (100%), distance 1 / 1.00 / 1, correct target 3927 (100%), correct head 3927 (100%)
2023-09-20 14:27:39.001 INFO  - Slot Event  *** Slot: 7366336, Block: 98e0d97275e7ab3d2d026479e3cd2f3878fe76a11cb05ac5fc7c8d0584c73cc9, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:27:51.003 INFO  - Slot Event  *** Slot: 7366337, Block: 6b11a2ca5c677fdd1db7fe846e727a1dd2314ac868d86cf5c970f23fb40c710d, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:28:03.001 INFO  - Slot Event  *** Slot: 7366338, Block: 9595ebbb7e0968c24cd9cc002096d165208374f2c9a24858ebb701a48882fbff, Justified: 230197, Finalized: 230196, Peers: 99
2023-09-20 14:28:15.000 INFO  - Slot Event  *** Slot: 7366339, Block: b38d973cfad6d46b1024d88f7e15ffa754579eb56fbdfdba825b6852c816010e, Justified: 230197, Finalized: 230196, Peers: 101
2023-09-20 14:28:27.001 INFO  - Slot Event  *** Slot: 7366340, Block: 1d47c5c7764aa836048294c6aa50560327a0a331a2875ff81711c1e0ef979ed9, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:28:39.000 INFO  - Slot Event  *** Slot: 7366341, Block: 484a03a9f44fc70fc6a4d78ccc8be12acccef52baeb170bbe8bf9b6c08eee92a, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:28:51.000 INFO  - Slot Event  *** Slot: 7366342, Block: cd44d36e5bc9119530d50a7a5cdc1266fceb9f92a479964eaaab43965b8cdd59, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:29:03.000 INFO  - Slot Event  *** Slot: 7366343, Block: 50084834c62433b636c16fd61bde57ca4c444d66021205f1b3f3452ad77176b6, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:29:15.000 INFO  - Slot Event  *** Slot: 7366344, Block: 6fbd384e0d03520a19a8cbaa876d4bf87503c4bfb418ac7f20eb4fc05e001adf, Justified: 230197, Finalized: 230196, Peers: 99
2023-09-20 14:29:27.001 INFO  - Slot Event  *** Slot: 7366345, Block: 156517367ddab7d4c78e2a2a17662ce091c5802bef076d8dac0e79fe81cc9b5f, Justified: 230197, Finalized: 230196, Peers: 99
2023-09-20 14:29:39.001 INFO  - Slot Event  *** Slot: 7366346, Block: 247f48e40f0e8032a9c87471c994c1f58699b5eeeb2b94c416fb390b6d21ea39, Justified: 230197, Finalized: 230196, Peers: 98
2023-09-20 14:29:51.000 INFO  - Slot Event  *** Slot: 7366347, Block: 193627d1eb4bdef23a7c99abbc251ceee485a0af6709b1fd930bc350d501f8b7, Justified: 230197, Finalized: 230196, Peers: 99
2023-09-20 14:30:03.000 INFO  - Slot Event  *** Slot: 7366348, Block: 3af293d03f911c801dbf72447db66b000ac1161843254a3ddd6d75b18d2d4c8a, Justified: 230197, Finalized: 230196, Peers: 99
2023-09-20 14:30:15.002 INFO  - Slot Event  *** Slot: 7366349, Block: cf9ba79c7442112eabfa0f097697f412e40e0e0c2ccc07a881a614151b45b08d, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:30:27.001 INFO  - Slot Event  *** Slot: 7366350, Block: 48533cf35ca4c6b8fef4ce19a6100b8faa995d32060bb15333682323f9d13d2d, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:30:39.000 INFO  - Slot Event  *** Slot: 7366351, Block: 1bafc96676081995dd07a093854e11f8de948435c4bf2c83c84be3ab6dd73fcd, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:30:51.000 INFO  - Slot Event  *** Slot: 7366352, Block: ee6dc0fc6b4562ae1a6146d1dfc8e8667906a92c5f81f9b12228515ae544510c, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:31:03.000 INFO  - Slot Event  *** Slot: 7366353, Block: d8467aa8d82a4aee97c342cbe2c2c1c43085fec121310175b470df7f39438f5a, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:31:11.001 INFO  - Updating number of persistent subnet subscriptions from 63 to 64
2023-09-20 14:31:15.001 INFO  - Slot Event  *** Slot: 7366354, Block: 8692caf59b0090b8e76a261a2f1dc719fe6588f9692a4db3520b344b4fad2792, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:31:27.001 INFO  - Slot Event  *** Slot: 7366355, Block: 187b702830b78dd414c0c38dbf325d818bfa5b6159b1b887c380f594cfa90799, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:31:39.070 INFO  - Slot Event  *** Slot: 7366356, Block: 5f14c6ccf8e41e0d0084727bb84049eca24707233c45f938088135aa21ffb24f, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:31:51.000 INFO  - Slot Event  *** Slot: 7366357, Block: 6d277550cd48484c30f972e8510f0d77acecfcf55c72814d23bfcba3d8985bee, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:32:03.001 INFO  - Slot Event  *** Slot: 7366358, Block: a0fec39abe79338947728f05373d390d73bcc7ecb3b7d46da7a141711c203a71, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:32:15.001 INFO  - Slot Event  *** Slot: 7366359, Block: 483ffa3d0eaf94bf6a30a14bf9a843b9228c1dab2f1449c8d44a940bd5a119cc, Justified: 230197, Finalized: 230196, Peers: 99
2023-09-20 14:32:27.000 INFO  - Slot Event  *** Slot: 7366360, Block: 93cea7f58eaedf729a359667c8eb9c9a23d38f783efddb73d9c518cf4f19c854, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:32:39.001 INFO  - Slot Event  *** Slot: 7366361, Block: 3bdabb7f2f746d8469e7af501043cf71b135b8fbf9fba521b9b2871502d507c4, Justified: 230197, Finalized: 230196, Peers: 101
2023-09-20 14:32:51.000 INFO  - Slot Event  *** Slot: 7366362, Block: 3b8c38844007d7de34c47e1f9add05035d82d181b56d39fde2d7275587698ab5, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:33:03.000 INFO  - Slot Event  *** Slot: 7366363, Block: 6f28712e90498587a8dfc5a38ccbc43715e9ea1fb66a1fcb7bb5c38933aea77f, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:33:15.201 INFO  - Calling local execution layer to start block production (block slot: 7366365)
2023-09-20 14:33:15.215 INFO  - Slot Event  *** Slot: 7366364, Block:                                                        ... empty, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:33:15.247 INFO  - Calling local execution layer to start block production (block slot: 7366365)
2023-09-20 14:33:15.268 WARN  - Late Block Import *** Block: bda42175a026df3c043010877bf0e148246c5cd4344885a5418a3370c1713e6d (7366364) proposer 621176 arrival 3962ms, pre-state_retrieved +5ms, processed +256ms, execution_payload_result_received +0ms, begin_importing +0ms, transaction_prepared +0ms, transaction_committed +0ms, completed +45ms
2023-09-20 14:33:23.004 INFO  - Creating unsigned block for slot 7366365
2023-09-20 14:33:24.151 INFO  - Received Builder Bid (Block Number = 18177721, Block Hash = 0x9e943925c32036fd34bfa216be6c066c8de32403651b73efd0402b9468eb22bc, MEV Reward (ETH) = 0.061677, Gas Limit = 30000000, Gas Used = 25477422)
2023-09-20 14:33:26.424 INFO  - Received execution payload from Builder (Block Number = 18177721, Block Hash = 0x9e943925c32036fd34bfa216be6c066c8de32403651b73efd0402b9468eb22bc)
2023-09-20 14:33:27.000 INFO  - Slot Event  *** Slot: 7366365, Block: f9fd2f955389b4ecdeb46225dda865df32ff07cfa5435858ff9fa4c938bfab8c, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:33:39.000 INFO  - Slot Event  *** Slot: 7366366, Block: fc422040810887f102eda54a8b44fca362fd5e29be428bed7b5904891946e50b, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:33:51.000 INFO  - Slot Event  *** Slot: 7366367, Block: 71c3c2425e8ff31709fc2b26ba0146e2451612b5baa0f7535afa2afef32c613a, Justified: 230197, Finalized: 230196, Peers: 100
2023-09-20 14:33:59.001 INFO  - Epoch Event *** Epoch: 230199, Justified checkpoint: 230198, Finalized checkpoint: 230197, Finalized root: 0b30e10efccf054528ec54f9c49c86382f955e5ce16342f52f82f0aea914514f
2023-09-20 14:33:59.001 INFO  - Block performance: epoch 230198, expected 1, produced 1, included 1 (100%)
2023-09-20 14:34:00.281 INFO  - Sync committee performance: epoch 230198, expected 96, produced 96, correct 93, included 93 (97%)
2023-09-20 14:34:00.387 INFO  - Attestation performance: epoch 230197, expected 3934, produced 3934, included 3803 (97%), distance 1 / 1.27 / 9, correct target 3807 (97%), correct head 3676 (93%)
2023-09-20 14:34:03.000 INFO  - Slot Event  *** Slot: 7366368, Block: 375f1359a0816b30de59d3f7205292e233bdcd3b7dca424b5a0ba5ae5e84a657, Justified: 230198, Finalized: 230197, Peers: 99
2023-09-20 14:34:15.001 INFO  - Slot Event  *** Slot: 7366369, Block: 8239758e1264eb8e99f34a53205469f14c5fe8251384bbf445884cb8d8caa461, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:34:27.001 INFO  - Slot Event  *** Slot: 7366370, Block: 870068fe9ae85b750be343d856f22c1c3e3b667a5afa22cc94c7cedab6918a30, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:34:39.000 INFO  - Slot Event  *** Slot: 7366371, Block: ed71348cb3c2d76ecdbbf0aece43692c56f369a945c892e47fd5be69a7f0b378, Justified: 230198, Finalized: 230197, Peers: 102
2023-09-20 14:34:51.001 INFO  - Slot Event  *** Slot: 7366372, Block: b074910aff588560621cd4848f82c39fa84ad32abe2185f7d8ef6963cedc231d, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:35:03.000 INFO  - Slot Event  *** Slot: 7366373, Block: 4d7070a4b5828d99fe669d9956387a4b574a1afe5b14b4a0faa028467f7052fe, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:35:15.000 INFO  - Slot Event  *** Slot: 7366374, Block: 2496f075a7dfd459737e9fec781c92865e229d418f7a99f96b8af7b9104efd7f, Justified: 230198, Finalized: 230197, Peers: 99
2023-09-20 14:35:27.000 INFO  - Slot Event  *** Slot: 7366375, Block: 4ddab70cc5f3a5b78d8f506c8a26d8cacf165ed3d3705ec5059f8fa1322e1995, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:35:39.001 INFO  - Slot Event  *** Slot: 7366376, Block: b925622332c9f08bee9fd322df6a154aa0e9f0e749d38f57f70bb06ad534f27c, Justified: 230198, Finalized: 230197, Peers: 99
2023-09-20 14:35:51.000 INFO  - Slot Event  *** Slot: 7366377, Block: d1643c36e22cd7b8a7cf2a13fba58f42b63b7e7dac42493cc1afbc26260d4a9c, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:36:03.000 INFO  - Slot Event  *** Slot: 7366378, Block: ea43fd79c87efbd213f108855a4ef7cae6a60656d765665418751cf53d4897d4, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:36:15.000 INFO  - Slot Event  *** Slot: 7366379, Block: a59e17cbe09a8d2a95de95329fb6878ebe25d7e7b912e379a64a4c277dedfeac, Justified: 230198, Finalized: 230197, Peers: 100
2023-09-20 14:36:27.000 INFO  - Slot Event  *** Slot: 7366380, Block: c8a9be268ac1e8f4a2ce68a168e61fd4be12ca4aeb2badede667bd4844d67f09, Justified: 230198, Finalized: 230197, Peers: 100
```

## Expected behaviour

No logs of this kind in the execution client

```bash
INFO [09-20|14:24:03.085] Starting work on payload                 id=0xb5a863509a649a2d
INFO [09-20|14:24:03.149] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=c7676a..428d56 txs=150  withdrawals=16 gas=13,572,373 fees=0.02325886562   root=f50817..afa9e2 elapsed=63.098ms
INFO [09-20|14:24:05.216] Updated payload                          id=0xb5a863509a649a2d number=18,177,675 hash=b82b0b..932f71 txs=161  withdrawals=16 gas=14,818,054 fees=0.02470676341   root=ceb3e8..b7f735 elapsed=66.760ms
```

**Comment by karalabe** [comment]
I'm not entirely sure what happened here. Geth was told to work on a block, so it does so. And it keeps regenerating the block every 3 seconds to pack better txs in. Eventually after nobody requests it for 12 seconds, the slot is missed so Geth just aborted maintaining the payload. That seems the correct behavior.

Now, I'm unsure why Teku detected the timeout. Geth clearly did get the payload construction request based on the logs, also Geth did not print any errors, so there's not much code where we could have timed out at that point:

```go
		payload, err := api.eth.Miner().BuildPayload(args)
		if err != nil {
			log.Error("Failed to build payload", "err", err)
			return valid(nil), engine.InvalidPayloadAttributes.With(err)
		}
		api.localBlocks.put(id, payload)
		return valid(&id), nil
```

**Comment by karalabe** [comment]
Do you see any other reports from Teku saying something timed out?

**Comment by holiman** [comment]
2023-09-20 14:24:03.084 INFO  - Calling local execution layer to start block production (block slot: 7366319)
INFO [09-20|14:24:03.085] Starting work on payload                 id=0xb5a863509a649a2d
... 5 seconds...
2023-09-20 14:24:08.002 WARN  - Execution Client request timed out. Make sure the Execution Client is online and can respond to requests.
2023-09-20 14:24:08.004 ERROR - Unable to import block: Execution Client returned an error: tech.pegasys.teku.ethereum.executionclient.response.InvalidRemoteResponseException: Invalid remote response from the execution client: TimeoutException
... 7 seconds ... 
INFO [09-20|14:24:15.086] Stopping work on payload                 id=0xb5a863509a649a2d reason=timeout
```
Teku says that Geth returned an error after 5 seconds, geth says that teku never came to collect the work.

**Comment by karalabe** [comment]
We've had an issue similar to this in the past, but people said it turned out to be a fault in Prysm: https://github.com/ethereum/go-ethereum/issues/27138

Unfortunately now it's a he-says-she-says thing. Something timed out between Geth/Teku, but there aren't any logs that could explain what/who.

@parithosh Did you notice any missed blocks with Geth on the various devnets?

**Comment by parithosh** [comment]
Hmm, no we're seeing proposals as expected: https://dencun-devnet-8.beaconcha.in/slots (both the proposer tags as well as the graffiti tell us who the proposer is). In general geth nodes seem healthy and are following the chain: https://forkmon.dencun-devnet-8.ethpandaops.io on the devnet.

**Comment by karalabe** [comment]
Seems Teku was somehow out of sync with the beacon chain before the messup started:

Teku logs say:

```
2023-09-20 14:24:03.083 INFO  - Slot Event  *** Slot: 7366318, Block:                                                        ... empty
```

But a block explorer says otherwise:

https://beaconcha.in/slot/7366318

**Comment by karalabe** [comment]
Teku does log later a late block, which matches the contents of the badly-reported-empty slot: 

```
Late Block Import *** Block: 992e19a9d1c8cc58fb88e39493ca2ec7370700b86b976dc43e7f82dcd74e71bb (7366318) proposer 377897 arrival 995ms, pre-state_retrieved +5ms, processed +165ms, execution_payload_result_received +7838ms, begin_importing +0ms, completed +1ms
```

**Comment by karalabe** [comment]
Ok, long story short, our best explanation is that Geth (in the old hash data scheme mode) will flush it's accumulated dirty trie data to disk every 3 days or so, which takes a huge disk IO hit, potentially taking a long time. Generally your node may fall a couple blocks behind during this time and then catch up.

Now, in this particular case, it would seem that you hit a block production event right when this flush was happening, and unfortunately, also when you were 1 block behind the beacon chain. That caused Teku to do weird requests and to report weird timeouts too.

With the old mode of operation of Geth, there's unfortunately no fix as this is just how it works. Annoyingly.

---

That said, Geth's path database scheme shipped in v1.13.x does do away with this very annoying point, so there should never be cases when Geth just gets stuck for a while doing "something".

Our only solution here is that we'd recommend setting up a path based Geth node and pointing Teku to that instead. If you'd rather wait a bit more until other people test it more, that's fine, we continually polish it up and release tiny fixes. When you do switch, just make sure to use the latest Geth release to have as many things fixed as we find.

Apologies about the missed block.

**Comment by preethiLN** [comment]
Hi,

We missed a block recently and these are the logs:

validator:
`2024-09-14T11:48:01.484Z
time="2024-09-14 11:48:01" level=error msg-"Failed to request block from beacon node" 
time=“2024-09-14 11:48:01" level=error msg="Failed to request block from beacon node" errors"rpc error: code = Internal desc = could not build block in parallel: rpc error: code = Internal desc = Could not get
local payload: could not get cached payload from execution client: timeout from http.Client" prefix=client
`

Beacon:
`time="2024-09-14 11:48:00" level=info msg="Begin building block" prefix="rpc/validator" sinceSlotStartTime=1.308675648s slot=9957538
time="2024-09-14 11:48:01" level=info msg="Finished building block" prefix="rpc/validator" sinceSlotStartTime=2.33806811s slot=9957538 validator=394095`

Geth:
`INFO [09-14|11:47:36.802] Chain head was updated                   number=20,748,644 hash=69bf28..6d6a5f root=e10172..939ea1 elapsed=2.120879ms
WARN [09-14|11:47:37.096] Ignoring already known beacon payload    number=20,748,644 hash=69bf28..6d6a5f age=2s
WARN [09-14|11:47:37.307] Ignoring already known beacon payload    number=20,748,644 hash=69bf28..6d6a5f age=2s
WARN [09-14|11:47:37.425] Ignoring already known beacon payload    number=20,748,644 hash=69bf28..6d6a5f age=2s
INFO [09-14|11:47:39.602] Ignoring beacon update to old head       number=20,748,643 hash=7d9ddf..3d72fa age=16s    have=20,748,644
INFO [09-14|11:47:49.157] Imported new potential chain segment     number=20,748,645 hash=4cc1ee..bcfd46 blocks=1 txs=326  mgas=16.645 elapsed=104.608ms   mgasps=159.119  snapdiffs=6.93MiB  triediffs=193.61MiB triedirty=45.38MiB
INFO [09-14|11:47:49.197] Chain head was updated                   number=20,748,645 hash=4cc1ee..bcfd46 root=4621fb..b722f1 elapsed=2.727655ms
WARN [09-14|11:47:49.623] Ignoring already known beacon payload    number=20,748,645 hash=4cc1ee..bcfd46 age=2s
WARN [09-14|11:47:49.855] Ignoring already known beacon payload    number=20,748,645 hash=4cc1ee..bcfd46 age=2s
WARN [09-14|11:47:49.893] Ignoring already known beacon payload    number=20,748,645 hash=4cc1ee..bcfd46 age=2s
INFO [09-14|11:47:50.561] Starting work on payload                 id=0x03458199dafeb3b3
INFO [09-14|11:47:50.578] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=c84456..f00a45 txs=27   withdrawals=16 gas=1,351,174  fees=0.001613999823  root=a81cbd..8f8c4f elapsed=17.446ms
INFO [09-14|11:47:52.602] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=07e8c7..c32ac6 txs=41   withdrawals=16 gas=2,608,100  fees=0.002181326173  root=d0c030..ef578d elapsed=22.651ms
INFO [09-14|11:47:54.629] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=485f7e..e31954 txs=57   withdrawals=16 gas=3,210,312  fees=0.002972246733  root=aea425..e13f47 elapsed=26.983ms
INFO [09-14|11:47:56.665] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=83036c..c2ed1e txs=89   withdrawals=16 gas=4,926,277  fees=0.003705007678  root=e4971f..7deebe elapsed=35.238ms
INFO [09-14|11:47:58.699] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=ecd898..db5514 txs=100  withdrawals=16 gas=6,003,465  fees=0.00426818459   root=1de852..7dc1a8 elapsed=34.263ms
INFO [09-14|11:48:00.408] Stopping work on payload                 id=0x03458199dafeb3b3 reason=delivery
INFO [09-14|11:48:13.365] Imported new potential chain segment     number=20,748,646 hash=73c553..e8b0a0 blocks=1 txs=208  mgas=13.104 elapsed=57.567ms    mgasps=227.625  snapdiffs=6.97MiB  triediffs=193.97MiB triedirty=45.69MiB
INFO [09-14|11:48:13.451] Chain head was updated                   number=20,748,646 hash=73c553..e8b0a0 root=369bcb..fe20c7 elapsed=1.827974ms
WARN [09-14|11:48:13.831] Ignoring already known beacon payload    number=20,748,646 hash=73c553..e8b0a0 age=2s
WARN [09-14|11:48:13.880] Ignoring already known beacon payload    number=20,748,646 hash=73c553..e8b0a0 age=2s
WARN [09-14|11:48:13.997] Ignoring already known beacon payload    number=20,748,646 hash=73c553..e8b0a0 age=2s
INFO [09-14|11:48:25.749] Imported new potential chain segment     number=20,748,647 hash=edad27..fa5d9b blocks=1 txs=119  mgas=8.517  elapsed=86.702ms    mgasps=98.231   snapdiffs=6.99MiB  triediffs=194.83MiB triedirty=45.73MiB
INFO [09-14|11:48:25.930] Chain head was updated                   number=20,748,647 hash=edad27..fa5d9b root=a5e574..bac431 elapsed=1.247306ms
`
MEV shows no errors during this time. We are on the latest version  Geth - 1.14.8, Prysm- 5.1.0. Any reason why we missed the block?

**Comment by holiman** [comment]
@preethiLN commenting on an old closed ticket is almost a guarantee that your comment will be overlooked, since we won't see it on our dashboards of open tickets. 

Please open a new issue for further communication. For now, though ,
```
INFO [09-14|11:47:50.578] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=c84456..f00a45 txs=27   withdrawals=16 gas=1,351,174  fees=0.001613999823  root=a81cbd..8f8c4f elapsed=17.446ms
INFO [09-14|11:47:52.602] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=07e8c7..c32ac6 txs=41   withdrawals=16 gas=2,608,100  fees=0.002181326173  root=d0c030..ef578d elapsed=22.651ms
INFO [09-14|11:47:54.629] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=485f7e..e31954 txs=57   withdrawals=16 gas=3,210,312  fees=0.002972246733  root=aea425..e13f47 elapsed=26.983ms
INFO [09-14|11:47:56.665] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=83036c..c2ed1e txs=89   withdrawals=16 gas=4,926,277  fees=0.003705007678  root=e4971f..7deebe elapsed=35.238ms
INFO [09-14|11:47:58.699] Updated payload                          id=0x03458199dafeb3b3 number=20,748,646 hash=ecd898..db5514 txs=100  withdrawals=16 gas=6,003,465  fees=0.00426818459   root=1de852..7dc1a8 elapsed=34.263ms
INFO [09-14|11:48:00.408] Stopping work on payload                 id=0x03458199dafeb3b3 reason=delivery
```
Geth spent about 10 seconds building a block, which was eventually delivered (collected by the CL). CL says there was som error, so might be better to start investigating there -- what it is that trigger that particular error message.
